beta/src/pages/learn/removing-effect-dependencies.md

```diff
@@ -0,0 +1,1286 @@
+}
+```
+
+And the linter would be right! Since `roomId` may change over time, this would introduce a bug in your code.
+
+**To remove a dependency, you need to "prove" to the linter that it *doesn't need* to be a dependency.** For example, you can move `roomId` out of your component to prove that it's not reactive and won't change on re-renders:
+
+```js {2,9}
+const serverUrl = 'https://localhost:1234';
+const roomId = 'music'; // Not a reactive value anymore
+
+function ChatRoom() {
+  useEffect(() => {
+    const connection = createConnection(serverUrl, roomId);
+    connection.connect();
+    return () => connection.disconnect();
+  }, []); // ✅ All dependencies declared
+  // ...
+}
+```
+
+Now that `roomId` is not a reactive value (and can't change on a re-render), it doesn't need to be a dependency:
+
+<Sandpack>
+
+```js
+import { useState, useEffect } from 'react';
+import { createConnection } from './chat.js';
+
+const serverUrl = 'https://localhost:1234';
+const roomId = 'music';
+
+export default function ChatRoom() {
+  useEffect(() => {
+    const connection = createConnection(serverUrl, roomId);
+    connection.connect();
+    return () => connection.disconnect();
+  }, []);
+  return <h1>Welcome to the {roomId} room!</h1>;
+}
+```
+
+```js chat.js
+export function createConnection(serverUrl, roomId) {
+  // A real implementation would actually connect to the server
+  return {
+    connect() {
+      console.log('✅ Connecting to "' + roomId + '" room at ' + serverUrl + '...');
+    },
+    disconnect() {
+      console.log('❌ Disconnected from "' + roomId + '" room at ' + serverUrl);
+    }
+  };
+}
+```
+
+```css
+input { display: block; margin-bottom: 20px; }
+button { margin-left: 10px; }
+```
+
+</Sandpack>
+
+This is why you could now specify an [empty (`[]`) dependency list.](/learn/lifecycle-of-reactive-effects#what-an-effect-with-empty-dependencies-means) Your Effect *really doesn't* depend on any reactive value anymore, so it *really doesn't* need to re-run when any of the component's props or state change.
+
+### To change the dependencies, change the code {/*to-change-the-dependencies-change-the-code*/}
+
+You might have noticed a pattern in your workflow:
+
+1. First, you **change the code** of your Effect or how your reactive values are declared.
+2. Then, you follow the linter and adjust the dependencies to **match the code you have changed.**
+3. If you're not happy with the list of dependencies, you **go back to the first step** (and change the code again).
+
+The last part is important. **If you want to change the dependencies, change the surrounding code first.** You can think of the dependency list as [a list of all the reactive values used by your Effect's code.](/learn/lifecycle-of-reactive-effects#react-verifies-that-you-specified-every-reactive-value-as-a-dependency) You don't intentionally *choose* what to put on that list. The list *describes* your code. To change the dependency list, change the code.
```

Comment:
> If you want to change the dependencies, change the surrounding code first.

Maybe this could be emphasized higher, maybe even in the introduction.

Reply:
yeah i've tried that before, and in practice people don't believe it even if it's the first thing on the page. i feel like it works better if you first show a few examples of the workflow, and *then* summarize the workflow.